### ISSUE TITLE
Error when trying to update contact/ECA address

### ISSUE THREAD

**Issue by freelock** [ledgersmb/LedgerSMB#2564]
Steps to reproduce: Search for existing vendor. Go to Addresses tab. Click Edit next to existing ECA address to change. (Page reloads and goes back to Company tab -- but Addresses tab now has address fields filled in). Change address line to correct address, and save. Result: Error

**Comment by freelock** [comment]
ERROR: duplicate key value violates unique constraint "eca_to_location_billing_u" DETAIL: Key (credit_id)=(536) already exists. CONTEXT: SQL statement "INSERT INTO eca_to_location (credit_id, location_class, location_id) VALUES (in_credit_id, in_location_class, l_id)" PL/pgSQL function eca__location_save(integer,integer,integer,text,text,text,text,text,text,integer,integer) line 39 at SQL statement: SELECT * FROM "public"."eca__location_save"(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) at /usr/local/lib/perl5/site_perl/5.24.0/PGObject.pm line 359.

**Comment by freelock** [comment]
Also "Delete" link does not delete the address -- there appears to be no way to change an address through the UI?

**Comment by ehuelsmann** [comment]
Given the mention of "billing_u", this should be tested with a billing address

**Comment by freelock** [comment]
I just updated a billing address on a credit account successfully in 1.8 successfully. (Have not tested Vendor recently).

Worked fine -- but one small usability improvement we should make:

To edit an existing address, you go to the "Addresses" tab and click "[Edit]" next to the address to edit. It populates the form fields below with the address, and allows you to save the location. However, it DOES NOT update the "Attach To'" dropdown, which shows the default "Entity" instead of the expected "Credit Account". I changed this before saving and it saved fine, but it should populate these dropdowns when you select an existing address to edit.
